### PR TITLE
Fix memory & goroutine leak

### LINKED ISSUE #20154: [Bug]: Memory & Goroutine leaks for tasks' context
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Environment

```markdown
- Milvus version:
- Deployment mode(standalone or cluster):
- SDK version(e.g. pymilvus v2.0.0rc2):
- OS(Ubuntu or CentOS): 
- CPU/Memory: 
- GPU: 
- Others:
```

### Current Behavior

The contexts of tasks which had never been added into scheduler, nerver be canceled

### Expected Behavior

_No response_

### Steps To Reproduce

_No response_

### Milvus Log

_No response_

### Anything else?

_No response_

### PR DESCRIPTION
**Pull request by yah01**
Signed-off-by: yah01 <yang.cen@zilliz.com>
/kind bug
fixes #20154

### CODE REVIEW

File: internal/querycoordv2/checkers/controller.go [hunk 1]
```diff
@@ -115,6 +115,7 @@ func (controller *CheckerController) check(ctx context.Context) {
 	for _, task := range tasks {
 		err := controller.scheduler.Add(task)
 		if err != nil {
+			task.Cancel()
```
Comment (soothing-rain) [inline internal/querycoordv2/checkers/controller.go:118]: https://github.com/milvus-io/milvus/blob/master/internal/querycoordv2/handlers.go#L126

Also here?

Comment (yah01) [reply internal/querycoordv2/checkers/controller.go:118]: Good sight, fixed

**Review by congqixia — APPROVED** [review]
/lgtm

**Pull request merged**